Add LaTex Support

moodyjmz opened the pull request:
### 📝 Summary

* Resolves: #6935 [<!-- Support Latex Math commands  -->](https://github.com/nextcloud/text/issues/6935)

This PR adds LaTeX math support to the text app.

it uses:

- Inline typing `$…$`
- Block typing `$$…$$`
- Pasting of inline
- Pasting of blocks
- Editing of both - I did consider using the same approach as inline links, the bubble, but decided to use the modal approach for both
- I added a feature into the menu to change existing content into maths - under the markup section

Code:
- I used a factory for making math nodes to avoid duplicating code
- It adds a modal, following the pattern of Image

Tests:
It adds some tests covering the features added, and issues I encountered while adding the feature

Limitations:

The existing modal doesn’t appear to be resizable - therefore here is set to largest


#### 🖼️ Screenshots

🏚️ Before | 🏡 After
---|---
<img width="77" height="38" alt="image" src="https://github.com/user-attachments/assets/39239cea-0356-441f-b813-2ffbe02e5ebf" />|<img width="99" height="34" alt="image" src="https://github.com/user-attachments/assets/df4fbd4b-5b57-4029-b572-025aeb088f4e" />
<img width="276" height="62" alt="image" src="https://github.com/user-attachments/assets/d387a39b-766d-4931-95bb-1dd97bb6242f" />|<img width="270" height="60" alt="image" src="https://github.com/user-attachments/assets/5a32569d-3124-4d61-91d6-261a254de781" />
<img width="227" height="294" alt="image" src="https://github.com/user-attachments/assets/54e7300d-12fc-4b42-bdbe-3d3dc5ec0f06" />|<img width="233" height="373" alt="image" src="https://github.com/user-attachments/assets/55dd0ff8-c08c-431b-bd73-26e7490c481b" />




### 🚧 TODO

- [ ] ...

### 🏁 Checklist

- [x] Code is properly formatted (`npm run lint` / `npm run stylelint` / `composer run cs:check`)
- [x] [Sign-off message](https://probot.github.io/apps/dco/) is added to all commits
- [x] [Tests](https://github.com/nextcloud/text#-testing-the-app) (unit, integration and/or end-to-end) passing and the changes are covered with tests
- [x] Documentation ([README](https://github.com/nextcloud/text/blob/main/README.md) or [documentation](https://github.com/nextcloud/documentation/blob/master/admin_manual/configuration_server/text_configuration.rst#L2)) has been updated or is not required

moodyjmz commented:
@max-nextcloud Sorry about the confusion, hopefully this PR is better to check. It only has one commit as I took the files from the other branch.

juliusknorr reviewed (commented):
Very nice, I left one inline comment and there are two CI checks that need some more work to pass.

When testing the modal seems to have an issue with not taking up the full height of the textarea, but I'm unsure if that is a component issue or could be addressed here as well. Would be fine to track as a follow up for polishing:

<img width="865" height="184" alt="Screenshot 2025-11-18 at 20 54 36" src="https://github.com/user-attachments/assets/302d3e4c-5b31-4239-8b3a-1f2e37db60d0" />

moodyjmz commented:
> Very nice, I left one inline comment and there are two CI checks that need some more work to pass.
> 
> When testing the modal seems to have an issue with not taking up the full height of the textarea, but I'm unsure if that is a component issue or could be addressed here as well. Would be fine to track as a follow up for polishing:
> 
> <img alt="Screenshot 2025-11-18 at 20 54 36" width="865" height="184" src="https://private-user-images.githubusercontent.com/3404133/515920003-302d3e4c-5b31-4239-8b3a-1f2e37db60d0.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjM0OTYyNTQsIm5iZiI6MTc2MzQ5NTk1NCwicGF0aCI6Ii8zNDA0MTMzLzUxNTkyMDAwMy0zMDJkM2U0Yy01YjMxLTQyMzktOGIzYS0xZjJlMzdkYjYwZDAucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MTExOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTExMThUMTk1OTE0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9M2MzZTA2MDNmNjJiMDZkZTBhNDYxOWM3Y2NkMjhlNmIxNmY1OTNiZTJjN2UyMjUyNzU4MjFkNDhkNDE2ZjExNSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.SLQBG1OLkNRjwr_FIUM_Fgznb9ZlNgRD4DS99Jw1zh8">

Thanks :) 

I addressed those issues.

Yes, the modal, I was unsure what caused it, but felt it was out of scope to address in this PR.

mejo- commented:
@moodyjmz thanks a lot for the contribution, much appreciated :heart: 

I just reviewed the PR (was on vacation last week) and realized that it adds `@tiptap/extension-mathematics` as NPM dependency but doesn't actually use it. I guess that it's a leftover from your development process? Also, looking at the upstream `@tiptap/extension-mathematics` code I wondered what was the reason why you decided to implement the whole mathematics node type yourself instead of using and extending the upstream `@tiptap/extension-mathematics` one :blush:
